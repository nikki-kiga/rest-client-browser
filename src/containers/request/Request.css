form {
    display: flex;
    flex-direction: column;
  }

    form div {
        display: flex;
        flex-direction: row;
        margin: 1rem;
    }
    form input[type=text] {
        background: #f5f5f5;
        border: none;
        border-bottom: 1px solid #000;
        font-size: 1.1rem;
        margin: 0;
        outline: none;
        padding: .25rem;
        transition: all 400ms ease-out;
        width: 80%;
      }

      form input[type=text]:focus {
        background: #fff;
        border-bottom: 2px solid blue;
      }
      .method {
        background: #f5f5f5;
        border: 1px solid #ccc;
        border-radius: 5px;
        cursor: pointer;
        margin-right: .5rem;
        padding: 0 1rem;
        line-height: 2rem;
        height: 2rem;

      }
      .method:hover {
        background: #ccc;
      }
      .submit{
        margin-left: 3rem;
      }

      textarea {
        flex: 1 1;
        height: 3rem;
        overflow-y: scroll;
    }
