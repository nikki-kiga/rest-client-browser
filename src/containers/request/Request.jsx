import React from 'react';
import PropTypes from 'prop-types';
import RESTButton from '../../components/RESTButton/RESTButton.jsx';
import Header from '../../components/Header/Header.jsx';
import styles from './Request.css';

const Request = ({ url, json, onChange, onSubmit }) => {
  const buttonTypes = ['get', 'post', 'put', 'delete'];
  const buttons = buttonTypes.map(type => (<RESTButton  key={type} reqType={type} onChange={onChange}/>));
  return (
    <>
      <Header/>
      <form onSubmit={onSubmit}>
        <div>
          <input type='text'name='url' value={url} onChange={onChange} placeholder='http://api.url.here'/>
          <button className={styles.submit}>Go!</button>
        </div>

        <div className={styles.methods}>
          {buttons}
          <textarea name='json' value={json} onChange={onChange}/>
        </div>
        
      </form>
    </>
  );
};

Request.propTypes = {
  url: PropTypes.string.isRequired,
  json: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Request;
