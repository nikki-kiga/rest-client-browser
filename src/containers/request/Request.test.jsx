import React from 'react';
import { shallow } from 'enzyme';
import Request from './Request.jsx';

describe('Request container', () => {
  let wrapper;
  let onSubmit;
  let handleChange;

  beforeEach(() => {
    onSubmit = jest.fn();
    handleChange = jest.fn();
        
    wrapper = shallow(<Request url='http://api.url.here' onChange={handleChange} onSubmit={onSubmit} />);
  });

  it('renders a request container', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
