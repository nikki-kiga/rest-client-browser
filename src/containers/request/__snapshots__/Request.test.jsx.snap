// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request container renders a request container 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    name="url"
    onChange={[MockFunction]}
    placeholder="http://api.url.here"
    value="http://api.url.here"
  />
  <button
    className="submit"
  >
    Go!
  </button>
  <Button
    key="get"
    onChange={[MockFunction]}
    reqType="get"
  />
  <Button
    key="post"
    onChange={[MockFunction]}
    reqType="post"
  />
  <Button
    key="put"
    onChange={[MockFunction]}
    reqType="put"
  />
  <Button
    key="delete"
    onChange={[MockFunction]}
    reqType="delete"
  />
  <input
    name="json"
    onChange={[MockFunction]}
  />
</form>
`;
